core/modules/comment/comment.entity.inc

```diff
@@ -505,7 +505,7 @@ class CommentStorageController extends EntityDatabaseStorageController {
       unset($comment->content);
 
       $build += array(
-        '#theme' => 'comment__node_' . $node->type . '__' . $view_mode,
+        '#theme' => 'comment__node_' . $node->type,
```

Comment:
Because you are still keeping `comment__node_TYPE` as the theme, this still takes precedence over `comment__node_TYPE__VIEW_MODE`.

<img width="457" alt="Screen Shot 2024-07-11 at 5 58 31 PM" src="https://github.com/user-attachments/assets/de45a250-70c5-470e-8e7f-0ca5fa919754">

<img width="1013" alt="Screen Shot 2024-07-11 at 5 58 48 PM" src="https://github.com/user-attachments/assets/790a20c3-981d-4dec-9586-8375b306269a">

In order to fix this and make the more specific template with the view mode take precedence, you need to modify:

`'#theme' => 'comment',`

AND then add the `comment__node_TYPE` to the suggestions below in the preprocess function.

Reply:
I wasn't sure about just using `comment` since it's also `comment_node_TYPE` in d7. Trying to walk the line of it still being a bug fix.

Reply:
I see. However, an important difference is that D7 does not provide the `comment--node-TYPE-VIEWMODE` suggestion, which avoids the problem I mentioned above.

So, then, perhaps we should not change anything related to `comment` at this point, since there is no bug. The code you have (adding the suggestion for `comment--node-TYPE-VIEWMODE`) actually creates a bug rather than solve it, since (with this PR) the `comment--node-TYPE`  template has preference over `comment--node-TYPE-VIEWMODE`

Reply:
I agree. I'll drop the changes to comment. I also ran into problems with File theme suggestions when trying to change it, so I also skipped that in this PR.